backend/src/main/java/com/ethan5/service/FriendService.java

```diff
@@ -19,4 +22,13 @@ public void addFriend(FriendRequest req) {
                 .build();
         repository.save(friendship);
     }
+
+    public void deleteFriend(String id1, String id2) {
+        Optional<Friend> f = repository.findByUser1IdAndUser2Id(id1, id2);
+        f.ifPresent(repository::delete);
+    }
+
+    public List<Friend> getFriendsList(String id) {
+        return repository.findByUser1Id(id);
```

Comment:
Here, we assume that friendships are represented bidirectionally. So, what we should do is when we create a new friend, we do the following:

`Friend friendship = Friend
                .builder()
                .user1Id(req.user1Id())
                .user2Id(req.user2Id())
                .build();
        repository.save(friendship);

        friendship = Friend
                .builder()
                .user1Id(req.user2Id())
                .user2Id(req.user1Id())
                .build();
    repository.save(friendship);
`

Reply:
You're right, adding in next commit